feat: redesign + add content (#3)
gnolang/.github: profile/README.md

<pre>
      ____ _____  ____      Discover:  <a href="https://www.gno.land/">Homepage</a> - <a href="https://github.com/gnolang/gno">Main repo</a>
     / __ `/ __ \/ __ \     Networks:  <a href="https://test3.gno.land">Test3 (latest)</a> - <a href="https://staging.gno.land">Staging</a>
    / /_/ / / / / /_/ /     Learn:     <a href="https://docs.gno.land">Official Documentation</a> - <a href="https://gno-by-example.com">Gno by Example</a> - <a href="https://github.com/gnolang/awesome-gno">Awesome Gno</a>
    \__, /_/ /_/\____/      Social:    <a href="https://twitter.com/_gnoland">Twitter</a> - <a href="https://discord.gg/tF2X8M6cVj">Discord</a> - <a href="https://reddit.com/r/gnoland">Reddit</a> - <a href="https://youtube.com/@_gnoland">YouTube</a>
   /____/                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</pre>

***

> <details><summary>Start</summary>

Reviewer: Not sure it should be collapsed

Reviewer 2: What do you mean, collapsed by default?

I'm not sure how else this would be done but we need the summary tags to encapsulate "Start" in order to prevent the default value "Details" being shown. Perhaps I am misunderstanding.

Reviewer: I suggest this:

```
## Getting Started

<details><summary>Why Gno?</summary>

...

</details>

<details>...
```

Reviewer 2: [7e63e94](https://github.com/gnolang/.github/pull/3/commits/7e63e94d78bfc4ad8d8f79b8be4f53c786c7e7b6)

**Learn how to write Smart Contracts with Gno**

- \[Quickstart] [I want to get write a simple contract in Gno](https://github.com/gnolang/gno/blob/master/docs/how-to-guides/simple-contract.md)

- \[Playground] [I want to learn Gnolang by writing and testing code on the Gno Playground](https://play.gno.land)

- \[Code-Editor Support] [I want to have a look at which code editors are supported for Gno development](https://github.com/gnolang/guide/blob/main/docs/engineering/gno.land/tools.md)

**Contributing**

- [I want to contribute to Gno](https://github.com/gnolang/gno/blob/master/CONTRIBUTING.md)

- [I want to understand more about your Grants & Ecosystem development](https://github.com/gnolang/ecosystem-fund-grants)

- [I want to learn more about incentivized contribution; ie, Game of Realms](https://gno.land/game-of-realms)
  
- [I want to see what other hackers are building on Gno.land](https://github.com/gnolang/hackerspace)

- [I want to contribute to your documentation efforts](https://github.com/gnolang/guide/blob/main/docs/engineering/conventions/documentation.md)

> - [I want to understand more about your code contribution best practices](https://github.com/gnolang/guide/blob/main/docs/engineering/conventions/go.md)

Reviewer 3: I love this `I want to...` format

    \__, /_/ /_/\____/      Social:    <a href="https://twitter.com/_gnoland">Twitter</a> - <a href="https://discord.gg/tF2X8M6cVj">Discord</a> - <a href="https://reddit.com/r/gnoland">Reddit</a> - <a href="https://youtube.com/@_gnoland">YouTube</a>
   /____/                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</pre>

***

## Why Gno? 

> - Interpreted Gnolang: Gno.land utilizes the Gnolang programming language, which is based on Go. It is executed through a specialized virtual machine called the GnoVM, purpose-built for the Gno.land blockchain. While Gnolang shares similarities with Go in terms of syntax, it currently lacks go routine support. However, this feature is planned for future development, ensuring deterministic GnoVM executions.

Reviewer: ```suggestion
- A better programming language for smart contracts: Gno.land utilizes the Gno programming language, which is an interpreted variation of Go. It is executed through a specialized virtual machine called the GnoVM, purpose-built for the Gno.land blockchain. There are some differences among the languages: Gno does not support Goroutines, channels and non-deterministic libraries like `crypto/rand`, and it supports additional features like automatic persistence of global variables for storing state.
```

> 

Reviewer: Bullet points in markdown don't need to be separated by empty lines

Reviewer 2: Yeah, I included these for readability since it does not impact the rendering

> - Consensus Protocol - Tendermint2: Gno.land achieves consensus between blockchain nodes using the Tendermint2 consensus protocol. This approach ensures secure and reliable network operation.

Reviewer: Okay, as a rule of thumb, use the question "Why is this better than ethereum / bitcoin?"

```suggestion
- Consensus Protocol: Gno.land achieves consensus between nodes using the Tendermint2 consensus algorithm. Tendermint2 is a variation on the well-established [Tendermint consensus algorithm](https://docs.tendermint.com/v0.34/introduction/what-is-tendermint.html) used in many Cosmos blockchains, which is Byzantine Fault Tolerant, has instant finality (Users can be sure their transactions are finalized as soon as a block is created, unlike Bitcoin) and can easily be extracted to create new interoperable TM2-based blockchains.
```

- Inter-Blockchain Communication (IBC): In the future, Gno.land will be able to communicate and exchange data with other blockchain networks within the Cosmos ecosystem through the Inter-Blockchain Communication (IBC) protocol.

## Why Go-Based?
The decision to base Gno.land's language on Go was influenced by the following factors:

> 
> - Standard and Secure Language: Go is a well-established and secure programming language, widely adopted in the software development community. By leveraging Go's features, Gno.land benefits from a robust and proven foundation.
> 
> - User-Friendly: Go's simplicity and ease of understanding make it beginner-friendly. This accessibility lowers the entry barrier for developers to create Smart Contracts on the Gno.land platform.

Reviewer: Please let's sync this question with the one on the [FAQ's](https://github.com/gnolang/awesome-gno/blob/main/FAQ.md), if not word-by-word at least in spirit. (That one can be improved too, but I'm just saying that we should combine the two things.)

```suggestion

- An established language: using Go instead of creating a new programming language enables existing Go developers to bring over their knowledge to Gno, making adopting the language not an intrinsic time investment on the success of the blockchain. Gno re-uses the syntax and a majority of the standard libraries, adapts Go's package model and adapts it for the blockchain context.
- Security through the package model: by leveraging Go's existing package model, which distinguishes between exported and unexported symbols, realms (smart contracts) are capable of simply discriminating what functionality, types and data should be available for consumption outside of the realm. By enforcing all on-chain published code to be open source and creating a public on-chain registry of published re-usable packages, Gno realms (smart contracts and packages can be easily audited and build upon each other's reputation.
```

Reviewer 2: Sure, though this PR predates the FAQ

Reviewer: A-ha, didn't see the date

> **Coming from other blockchains like Ethereum?**

Reviewer: ```suggestion
## Coming from other blockchains like Ethereum?
```

Gno.land offers a few distinct advantages to Ethereum:

1. Transparent and Auditable Smart Contracts: Gno.land Smart Contracts are fully transparent and auditable by users because the actual source code is uploaded to the blockchain. In contrast, Ethereum uses Solidity, which requires precompilation into bytecode, leading to less transparency as bytecode is stored on the blockchain, not the human-readable source code.

> 2. General-Purpose Language: Gno.land's Gnolang is a general-purpose language, similar to Go, extending its usability beyond the context of blockchain. In contrast, Solidity is designed specifically for Smart Contracts on the Ethereum platform.

Reviewer: ```suggestion
2. General-Purpose Language: Gno.land's Gno is a general-purpose language, similar to Go, extending its usability beyond the context of blockchain, and allowing Gno to tap into the already existing ecosystem of Go packages. In contrast, Solidity is designed specifically for Smart Contracts on the Ethereum platform.
```
